sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -304,33 +308,78 @@ export class ManagementClient extends LinkEntity {
   private async _makeManagementRequest(
     request: Message,
     options?: { retryOptions?: RetryOptions; abortSignal?: AbortSignalLike; requestName?: string }
-  ): Promise<any> {
+  ): Promise<any> { 
     try {
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
-
       if (!options) {
         options = {};
       }
 
-      const sendRequestOptions: SendRequestOptions = {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+
+          let waitTimer: any;
+          const actionAfterTimeout = () => {
+            const address = this._mgmtReqResLink!.receiver.address || "address";
+            const desc: string =
+              `The request with message_id "${request.message_id}" to "${address}" ` +
+              `endpoint timed out. Please try again later.`;
+            const e: Error = {
+              name: "OperationTimeoutError",
+              message: desc
+            };
+            return reject(translate(e));
+          };
+
+          waitTimer = setTimeout(
+            actionAfterTimeout,
+            getRetryAttemptTimeoutInMs(options!.retryOptions)
+          );
+
+          const initOperationStartTime = Date.now();
+          log.mgmt(
+            "[%s] Acquiring lock to get the management req res link.",
+            this._context.connectionId
+          );
+          await defaultLock.acquire(this.managementLock, () => {
+            return this._init().catch((err: Error) => {
+              clearTimeout(waitTimer);
+              reject(translate(err));
+            });
+          });
+          const initOperationEndTime = Date.now();
+
+          if (!options) {
```

Comment:
This code already exists in the outer scope (and can be removed if you use a default assignment in the function signature)